mountpoint-s3/src/inode.rs

```diff
@@ -127,7 +126,7 @@ impl Superblock {
             return Err(InodeError::NotADirectory(parent_ino));
         }
         let mut full_path = parent.full_key().to_owned();
-        assert!(full_path.is_empty() || full_path.ends_with('/'));
+        assert!(Prefix::is_valid(&full_path));
```

Comment:
I wouldn't change this—to me they feel similar only coincidentally; a `Prefix` is "a subset of the bucket" whereas this is more like "a query I'm about to send to S3". I can imagine worlds where we might change the behavior of `Prefix` in the future (for example, to no longer require ending in a `/`).